src/services/expensesService.js

```diff
@@ -0,0 +1,54 @@
+let expenses = [];
+
+const resetExpenses = () => {
+  expenses = [];
+};
+
+const getAllExpenses = (userId, categories, from, to) => {
+  return expenses
+    .filter((expense) => !userId || expense.userId === +userId)
+    .filter((expense) => !categories || expense.category === categories)
+    .filter((expense) => !from || new Date(expense.spentAt) >= new Date(from))
+    .filter((expense) => !to || new Date(expense.spentAt) <= new Date(to));
+};
+
+const getExpenseById = (id) => {
+  return expenses.find((expense) => expense.id === +id);
+};
+
+const createExpense = (userId, spentAt, title, amount, category, note) => {
+  const expense = {
+    id: Math.floor(Math.random() * 1000),
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  };
```

Comment:
The id for the expense is generated using `Math.random()`. While this works for small applications, it may create conflicts in larger applications where the number of expenses is large. Consider using a more reliable method for generating unique ids, such as a UUID generator.